src/main/java/ru/test/restservice/service/ProjectService.java

```diff
@@ -28,69 +33,98 @@
-        projs.addObject("projects",
-                projectRepository
-                        .findAll()
-                        .stream()
-                        .filter(pr -> pr.owners.contains(user))
-                        .collect(Collectors.toList()));
-        return projs;
+    public List<Project> listProjects(String username) {
+        User user = userRepository.findByEmail(username);
+        return projectRepository
+                .findAll()
+                .stream()
+                .filter(pr -> pr.owners.contains(user))
+                .collect(Collectors.toList());
     }
 
-    public void createNewProject(String projectName, Authentication auth) throws IOException {
-        User currentUser = userRepository.findByEmail(auth.getName());
+    public void createNewProject(String projectName, String email) throws IOException, GitAPIException {
+        User currentUser = userRepository.findByEmail(email);
         UUID newId = UUID.randomUUID();
         Project newProj = new Project(newId, projectName, "projects/" + newId, Collections.singleton(currentUser));
         projectRepository.save(newProj);
         Files.createDirectories(Paths.get("projects/" + newId));
         Files.createFile(Paths.get("projects/" + newId + "/" + projectName + ".tex"));
+        Git git = Git.init().setDirectory(Paths.get("projects/" + newId).toFile()).call();
+        git.add().addFilepattern(".").call();
+        git.commit().setMessage("Initial commit").call();
     }
 
-    public void deleteProject(UUID projectId) throws IOException {
-        Project projectToDelete = projectRepository.findById(projectId).get();
-        FileSystemUtils.deleteRecursively(Paths.get(projectToDelete.path));
+    public void deleteProject(UUID projectId) {
+        Project projectToDelete = projectRepository.findById(projectId).orElseThrow(NotFoundException::new);
+        FileSystemUtils.deleteRecursively(Paths.get(projectToDelete.path).toFile());
         projectRepository.delete(projectToDelete);
     }
 
     public void checkOwnership(UUID projectId, String email) {
-        Optional<Project> project = projectRepository.findById(projectId);
+        Project project = projectRepository.findById(projectId).orElseThrow(NotFoundException::new);
         User user = userRepository.findByEmail(email);
-        if (user == null || !project.isPresent() || !project.get().owners.contains(user)) {
+        if (user == null || !project.owners.contains(user)) {
             throw new NotFoundException();
         }
     }
 
     // TODO: добавить виды прав?
     // TODO: сообщение для страницы ошибки
     // TODO: удалять проект, если не осталось владельцев
+    // TODO: блокирование проекта для одновременной работы
 
     public void addOwner(UUID projectId, String email) {
-        Optional<Project> project = projectRepository.findById(projectId);
+        Project project = projectRepository.findById(projectId).orElseThrow(NotFoundException::new);
```

Comment:
Вообще принято вот так оформлять:
```suggestion
        Project project = projectRepository.findById(projectId)
                                             .orElseThrow(NotFoundException::new);
```